### PR TITLE
Add GitHub actions

### PR DESCRIPTION
**Pull request by dynobo**

### CODE REVIEW

**Review by LGro — APPROVED** [review]
:rocket:

**Comment by dynobo** [comment]
For the tests to pass, we need [parameterspace PR#19](https://github.com/boschresearch/parameterspace/pull/19).

File: poetry.lock [hunk 1]
```diff
@@ -0,0 +1,1631 @@
+[[package]]
```
Comment (dynobo) [inline poetry.lock:1]: - [ ] Before merging, do `poetry update` (requires [parameterspace #19](https://github.com/boschresearch/parameterspace/pull/19))

Comment (LGro) [reply poetry.lock:1]: I'm afraid it's not as simple as adding `__iter__`. What this linting issue highlights is that BOHB is not ok with any `SearchSpace` but actually requires a `ParameterSpace`.
I'd recommend to mark the current linting issue as skip and create an issue for us to figure out separately from this PR how to deal with optimizers that requires specific knowledge about the parameter types. Maybe we need to add them to the base `SearchSpace` implementation in some way as well or allow individual optimizers to be more specific about the search space definition they allow.

Comment (dynobo) [reply poetry.lock:1]: Created issue #10 for this.
I will set the line to be ignored by pylint. We should re-enable that, after the issue is fixed.

**Comment by dynobo** [comment]
The solution will be discussed and resolved as part of `blackboxopt` [issue 10](https://github.com/boschresearch/blackboxopt/issues/10)

**Pull request merged**